src/main.py

```diff
@@ -0,0 +1,117 @@
+from alerts_aggregator import AlertsAggregator
+from oref_alert_indexer import OrefAlertsIndexer  
+from fastapi import FastAPI, HTTPException , Depends, Header
+import pandas as pd
+import requests
+from errors import OrefAPIException, WrongSettlementException, NoAlarmsException, InvalidSettlement
+from models import AlertsQueryInput
+from datetime import datetime, timedelta
+app = FastAPI()
+
+CSV_FILE_PATH = "C:/Projects/Python/Projects/BestTimeToShower/data/data.csv"
+dataframe = pd.read_csv(CSV_FILE_PATH)
+alert_indexer = OrefAlertsIndexer()
+alert_aggregator = AlertsAggregator(dataframe)
+
+def get_api_key(api_key: str = Header(..., convert_underscores=False)):
+
+    if api_key == "red_color":
```

Comment:
call it `ADMIN_API_KEY` . and store the actual key (`"red_color"`) in .evn file not in your code (!)